packages/ra-ui-materialui/src/input/ReferenceArrayInput.spec.tsx

```diff
@@ -244,4 +249,52 @@ describe('<ReferenceArrayInput />', () => {
             });
         });
     });
+
+    it('should show selected values when ids type are inconsistant', async () => {
+        render(<HandlingTypesDiscrepencies />);
+        await waitFor(() => {
+            expect(
+                screen.queryByText('#1', {
+                    selector: 'div.MuiChip-root .MuiChip-label',
+                })
+            ).not.toBeNull();
+        });
+        expect(
+            screen.queryByText('#2', {
+                selector: 'div.MuiChip-root .MuiChip-label',
+            })
+        ).not.toBeNull();
+        expect(
+            screen.queryByText('artist_3', { selector: 'div.MuiChip-root' })
+        ).toBeNull();
+    });
+
+    it('should unselect a value whose id type is inconsistant', async () => {
+        render(<HandlingTypesDiscrepencies />);
+
+        const unselect = (queriedText: string) => () => {
+            const chip = screen.queryByText(queriedText, {
+                selector: '.MuiChip-label',
+            })?.nextSibling;
```

Comment:
if you expect the element to be there, you should use `getByText` instead of `queryByText`

Have a look at [this cheatsheet](https://testing-library.com/docs/react-testing-library/cheatsheet/#queries) to learn more about the differences.